cmd/scanner/scanner.go

```diff
@@ -180,152 +185,174 @@ func (s *Scanner) createStockers() error {
 
 // startStockers runs every stocker instance
 func (s *Scanner) startStockers() error {
+	var wg sync.WaitGroup
+	done := make(chan struct{})
+	errChan := make(chan error, len(s.stockers))
+
+	// Running Stockers.MakeStock procedure on parallel
 	for _, stockerInstance := range s.stockers {
-		err := stockerInstance.MakeStock()
-		if err != nil {
-			return err
-		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			if err := stockerInstance.MakeStock(); err != nil {
+				errChan <- err
+			}
+		}()
+	}
+
+	// Waiting for every Stock
+	go func() {
+		wg.Wait()
+		close(done)
+	}()
+
+	// Processing errors when every stocker has finished
+	select {
+	case <-done:
+		close(errChan)
+	case err := <-errChan:
```

Comment:
I think we have a bug where we only catch one error and the others are dropped. I've created a simple proof in the Go Playground that demonstrates this exact problem using our logic

https://go.dev/play/p/NRoimoCPwaA?v=goprev